Task/config options doc (#853)
LLNL/axom: src/docs/sphinx/coding_guide/sec10_dev_macros.rst

      // rest of method implementation
   }

Axom provides various other macro constants for conditionally-compiled code 

> based on which built-in and third-party libraries are being used and which Axom 

Reviewer: Incremental improvement?
```suggestion
that reflect which built-in and third-party libraries are being used and which Axom 
```

Author: changed.
